include/nn_dot_product_matrix.h

```diff
@@ -0,0 +1,28 @@
+#ifndef NN_DOT_PRODUCT_MATRIX_H
+#define NN_DOT_PRODUCT_MATRIX_H
+
+#include "nn_error.h"
+#include <stddef.h>
+
+// MATRIX_MAX_ROWS defines the maximum number of rows in a matrix.
+#ifndef MATRIX_MAX_ROWS
+#define MATRIX_MAX_ROWS 3
```

Comment:
We should add prefix to all defines since this code base includes other libraries (https://github.com/devfacet/nntbn/tree/main/lib) and itself will be included by other libs/apps. We don't want conflict.

See `NN_DOT_PROD_` prefix in https://github.com/devfacet/nntbn/blob/main/include/nn_dot_product.h